### PR TITLE
Upgrade deps, and add some more eslint rules

### PR DESCRIPTION
**Pull request by bhollis**

### CODE REVIEW

**Comment by sundevour** [comment]
love these. interesting bug in travis though..

**Comment by bhollis** [comment]
Something about the Jest upgrade may be messing it up

File: yarn.lock [hunk 1]
```diff
@@ -9143,6 +9124,16 @@ pretty-format@^25.1.0, pretty-format@^25.2.1, pretty-format@^25.5.0:
     ansi-styles "^4.0.0"
     react-is "^16.12.0"
 
+pretty-format@^26.0.1:
+  version "26.0.1"
+  resolved "https://registry.yarnpkg.com/pretty-format/-/pretty-format-26.0.1.tgz#a4fe54fe428ad2fd3413ca6bbd1ec8c2e277e197"
+  integrity sha512-SWxz6MbupT3ZSlL0Po4WF/KujhQaVehijR2blyRDCzk9e45EaYMVhMBn49fnRuHxtkSpXTes1GxNpVmH86Bxfw==
+  dependencies:
+    "@jest/types" "^26.0.1"
```
Comment (delphiactual) [inline yarn.lock:9132]: 26

File: yarn.lock [hunk 2]
```diff
@@ -1175,6 +1168,16 @@
     "@types/yargs" "^15.0.0"
     chalk "^3.0.0"
 
+"@jest/types@^26.0.1":
```
Comment (delphiactual) [inline yarn.lock:1171]: 26

**Pull request merged**